## tl;dr

当前可执行的纯自产 Exchange 排名中，Fodder 第一；Pumpkin 的纸面利润更高，但属于 AutumnHarvest 季节品，当前排名剔除。

## Context & Methods

The analysis reads the live state, measured production rates and wages, the rolling ticker, the rotating order-book volume estimate, and the measured recipe graph. It deducts the 4% exchange fee and outbound transportation. Inputs are recursively self-produced, and throughput is capped by the first one-building bottleneck.

### Key Assumptions

- Seasonal production is excluded from the executable ranking.
- Quality is Q0 because the current inventory and quotes are Q0.
- The volume estimate includes order cancellations and is supporting liquidity evidence, not guaranteed purchases.
- Rankings are per self-contained one-building production line, so products are compared on like-for-like productive capacity.


In [1]:
from analysis import build_analysis

analysis = build_analysis()
for row in analysis["current_ranking"][:8]:
    print({key: row[key] for key in ("rank", "product", "sustainable_profit_per_hour", "sustainable_output_per_hour", "bottleneck")})


[
  {
    "rank": 1,
    "product": "Fodder",
    "profit_per_hour": 468.9,
    "output_per_hour": 108.62,
    "bottleneck": "Farm"
  },
  {
    "rank": 2,
    "product": "Grapes",
    "profit_per_hour": 408.7,
    "output_per_hour": 404.79,
    "bottleneck": "Water reservoir"
  },
  {
    "rank": 3,
    "product": "Grain",
    "profit_per_hour": 395.7,
    "output_per_hour": 1296.84,
    "bottleneck": "Farm"
  },
  {
    "rank": 4,
    "product": "Cocoa Beans",
    "profit_per_hour": 392.2,
    "output_per_hour": 436.57,
    "bottleneck": "Farm"
  },
  {
    "rank": 5,
    "product": "Cotton",
    "profit_per_hour": 383.4,
    "output_per_hour": 613.71,
    "bottleneck": "Farm"
  },
  {
    "rank": 6,
    "product": "Flour",
    "profit_per_hour": 342.8,
    "output_per_hour": 86.46,
    "bottleneck": "Farm"
  },
  {
    "rank": 7,
    "product": "Oranges",
    "profit_per_hour": 318.5,
    "output_per_hour": 470.96,
    "bottleneck": "Farm"
  },
  {
    "rank": 8,
    "product": "App

## Results

Fodder wins the currently executable ranking because its exchange revenue covers the full Grain + Vegetables + Seeds + Water + Power chain and Mill labor. The L3 Farm is the bottleneck, so one self-sufficient line produces about 109 Fodder per hour even though an L1 Mill can process much more.


In [2]:
print(analysis["quality_checks"])


{
  "candidate_count": 17,
  "current_candidate_count": 16,
  "all_candidates_have_prices": true,
  "all_candidates_have_positive_rates": true,
  "seasonal_products_excluded": [
    "Pumpkin"
  ],
  "top_product": "Fodder",
  "top_remains_positive_at_24h_p10": true,
  "top_demand_exceeds_sustainable_output": true
}


## Takeaways

- Use Fodder as the Exchange-only benchmark, not as an automatic company-wide pivot.
- Coffee Powder remains valuable for Grocery retail and benefits from the company's unusually large Mill fleet.
- Re-rank at Pumpkin season, after production modifiers expire, or after a material price move.
